lib/tvdbapiv2/rest.py

```diff
@@ -51,125 +31,50 @@
         """
-        return self.urllib3_response.getheaders()
+        return self.requests_response.headers
 
     def getheader(self, name, default=None):
         """
         Returns a given response header.
         """
-        return self.urllib3_response.getheader(name, default)
+        return self.requests_response.headers.get(name, default)
 
 
 class RESTClientObject(object):
 
-    def __init__(self, pools_size=4):
-        # urllib3.PoolManager will pass all kw parameters to connectionpool
-        # https://github.com/shazow/urllib3/blob/f9409436f83aeb79fbaf090181cd81b784f1b8ce/urllib3/poolmanager.py#L75
-        # https://github.com/shazow/urllib3/blob/f9409436f83aeb79fbaf090181cd81b784f1b8ce/urllib3/connectionpool.py#L680
-        # ca_certs vs cert_file vs key_file
-        # http://stackoverflow.com/a/23957365/2985775
-
-        # cert_reqs
-        if Configuration().verify_ssl:
-            cert_reqs = ssl.CERT_REQUIRED
-        else:
-            cert_reqs = ssl.CERT_NONE
-
-        # ca_certs
-        if Configuration().ssl_ca_cert:
-            ca_certs = Configuration().ssl_ca_cert
-        else:
-            # if not set certificate file, use Mozilla's root certificates.
-            ca_certs = certifi.where()
-
-        # cert_file
-        cert_file = Configuration().cert_file
-
-        # key file
-        key_file = Configuration().key_file
-
-        # https pool manager
-        self.pool_manager = urllib3.PoolManager(
-            num_pools=pools_size,
-            cert_reqs=cert_reqs,
-            ca_certs=ca_certs,
-            cert_file=cert_file,
-            key_file=key_file
-        )
-
-    def request(self, method, url, query_params=None, headers=None,
-                body=None, post_params=None):
-        """
-        :param method: http request method
-        :param url: http request url
-        :param query_params: query parameters in the url
-        :param headers: http request headers
-        :param body: request json body, for `application/json`
-        :param post_params: request post parameters,
-                            `application/x-www-form-urlencode`
-                            and `multipart/form-data`
-        """
-        method = method.upper()
-        assert method in ['GET', 'HEAD', 'DELETE', 'POST', 'PUT', 'PATCH', 'OPTIONS']
-
-        if post_params and body:
-            raise ValueError(
-                "body parameter cannot be used with post_params parameter."
-            )
+    def __init__(self, session=None):
+        self.session = session
 
-        post_params = post_params or {}
-        headers = headers or {}
-
-        if 'Content-Type' not in headers:
-            headers['Content-Type'] = 'application/json'
+    def request(self, method, url, query_params=None, headers=None, body=None, post_params=None):
 
         try:
             # For `POST`, `PUT`, `PATCH`, `OPTIONS`
             if method in ['POST', 'PUT', 'PATCH', 'OPTIONS']:
```

Comment:
I think I changed everything except the if .. else. I don't really see the need for changing this right now. Can you live with that?